src/atoms/SegmentControl/__tests__/SegmentControl.native.test.js

```diff
@@ -0,0 +1,478 @@
+            Option 1
+          </SegmentControl.Option>
+          <SegmentControl.Option value="2" testID="segment-option-2">
+            Option 2
+          </SegmentControl.Option>
+          <SegmentControl.Option value="3" testID="segment-option-3">
+            Option 3
+          </SegmentControl.Option>
+        </SegmentControl>,
+      );
+      const option = getByTestId('segment-option-1');
+
+      act(() => {
+        fireEvent.pressIn(option);
+      });
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('onPress', () => {
+    it('outlined segment control ', () => {
+      const { container, getByTestId } = renderWithTheme(
+        <SegmentControl defaultValue="1" variant="outlined" size="small">
+          <SegmentControl.Option value="1" testID="segment-option-1">
+            Option 1
+          </SegmentControl.Option>
+          <SegmentControl.Option value="2" testID="segment-option-2">
+            Option 2
+          </SegmentControl.Option>
+          <SegmentControl.Option value="3" testID="segment-option-3">
+            Option 3
+          </SegmentControl.Option>
+        </SegmentControl>,
+      );
+      const option = getByTestId('segment-option-2');
+
+      act(() => {
+        fireEvent.press(option);
+      });
+      expect(container).toMatchSnapshot();
+    });
+    it('filled segment control ', () => {
+      const { container, getByTestId } = renderWithTheme(
+        <SegmentControl defaultValue="1" variant="filled" size="small">
+          <SegmentControl.Option value="1" testID="segment-option-1">
+            Option 1
+          </SegmentControl.Option>
+          <SegmentControl.Option value="2" testID="segment-option-2">
+            Option 2
+          </SegmentControl.Option>
+          <SegmentControl.Option value="3" testID="segment-option-3">
+            Option 3
+          </SegmentControl.Option>
+        </SegmentControl>,
+      );
+      const option = getByTestId('segment-option-2');
+
+      act(() => {
+        fireEvent.press(option);
+      });
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('disabled', () => {
+    it('renders disabled outlined segment control ', () => {
+      const { container } = renderWithTheme(
+        <SegmentControl defaultValue="1" variant="outlined">
+          <SegmentControl.Option value="1" disabled testID="segment-option-1">
+            Option 1
+          </SegmentControl.Option>
+          <SegmentControl.Option value="2" disabled testID="segment-option-2">
+            Option 2
+          </SegmentControl.Option>
+          <SegmentControl.Option value="3" disabled testID="segment-option-3">
+            Option 3
+          </SegmentControl.Option>
+        </SegmentControl>,
+      );
+      expect(container).toMatchSnapshot();
+    });
+
+    it('renders disabled filled segment control ', () => {
+      const { container } = renderWithTheme(
```

Comment:
there should be a test case that renders a segment control with few options disabled. Right now I can just see where all the options are disabled.